keystore/kms/base.go

```diff
@@ -0,0 +1,77 @@
+package kms
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"sync"
+)
+
+var lock = sync.Mutex{}
+var encryptorCreators = map[string]EncryptorCreateFunc{}
+
+// EncryptorCreateFunc generic function for creating Encryptor
+type EncryptorCreateFunc func(credentialPath string) (Encryptor, error)
+
+// RegisterEncryptorCreator add new EncryptorCreator to registry
+func RegisterEncryptorCreator(encryptorID string, encryptorCreateFunc EncryptorCreateFunc) {
+	lock.Lock()
+	encryptorCreators[encryptorID] = encryptorCreateFunc
```

Comment:
hm. maybe lets log debug info about the registered creator? probably it will help debug issues with unsupported modules in the future

Reply:
Agree!